build/UserALEWebExtension/background.js

```diff
@@ -1131,28 +1155,48 @@ var defaultConfig = {
     authHeader: null,
     toolName: 'useralePlugin',
     version: version
+  },
+  pluginConfig: {
+    // Default to a regex that will match no string
+    urlWhitelist: '(?!x)x'
   }
 };
-browser.storage.local.get(defaultConfig, function (res) {
-  options(res.useraleConfig);
-});
+var urlWhitelist;
+function updateConfig(config) {
+  urlWhitelist = new RegExp(config.pluginConfig.urlWhitelist);
+  options(config.useraleConfig);
+  dispatchTabMessage(config.useraleConfig);
+}
 function dispatchTabMessage(message) {
   browser.tabs.query({}, function (tabs) {
     tabs.forEach(function (tab) {
       browser.tabs.sendMessage(tab.id, message);
     });
   });
 }
+function filterUrl(log) {
```

Comment:
Code smell to have two different return types depending on conditional. Since you're not actually modifying the log, you don't need to return the log.

Better to have function name like `isUrlWhitelisted` which only returns bool.

Reply:
This function is a little wonky because it has to match the pattern expected by addcallbacks(). Callbacks are called by package log functions and modify logs or drop them if given a falsey value. I've already created a ticket to change this callback functionality to match the pattern you used for modifying headers. Once the core api is changed, we can modify how we're using it here.

Here's some background on why addcallbacks is the way it is:
https://github.com/apache/flagon-useralejs/issues/312
https://github.com/apache/flagon-useralejs/pull/350

Reply:
Understood.